test/integration/limited_ciphers_utils_test.go

```diff
@@ -0,0 +1,155 @@
+	require.NoError(t, err)
+
+	// As a precondition of this test, ensure that the list of allowedCiphers is empty
+	require.Empty(t, config.TLS.OneDotTwo.AllowedCiphers)
+
+	config.TLS.OneDotTwo.AllowedCiphers = allowedCiphers
+
+	updatedSupervisorConfig, err := yaml.Marshal(config)
+	require.NoError(t, err)
+
+	supervisorStaticConfigMap.Data = make(map[string]string)
+	supervisorStaticConfigMap.Data["pinniped.yaml"] = string(updatedSupervisorConfig)
+
+	_, err = configMapClient.Update(ctx, supervisorStaticConfigMap, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	// this will cycle all the pods
+	restartAllPodsOfApp(t, env.SupervisorNamespace, env.SupervisorAppName, false)
+
+	startKubectlPortForward(ctx, t, "10509", "443", env.SupervisorAppName+"-nodeport", env.SupervisorNamespace)
+	stdout, stderr := testlib.RunNmapSSLEnum(t, "127.0.0.1", 10509)
+	require.Empty(t, stderr)
+
+	expectedCiphersConfig := &tls.Config{
+		MinVersion:   tls.VersionTLS12,
+		MaxVersion:   testlib.MaxTLSVersion,
+		CipherSuites: expectedCiphers,
+	}
+
+	require.Contains(t, stdout, testlib.GetExpectedCiphers(expectedCiphersConfig, "server"), "stdout:\n%s", stdout)
+}
+
+// restartAllPodsOfApp will immediately scale to 0 and then scale back.
+// There are no uses of t.Cleanup since these actions need to happen immediately.
+func restartAllPodsOfApp(
+	t *testing.T,
+	namespace string,
+	appName string,
+	isConcierge bool,
+) {
+	t.Helper()
+
+	ignorePodsWithNameSubstring := ""
+	if isConcierge {
+		ignorePodsWithNameSubstring = "-kube-cert-agent-"
+	}
+
+	// Precondition: the app should have some pods running initially.
+	initialPods := getRunningPodsByNamePrefix(t, namespace, appName+"-", ignorePodsWithNameSubstring)
+	require.Greater(t, len(initialPods), 0)
+
+	// Scale down the deployment's number of replicas to 0, which will shut down all the pods.
+	originalScale := updateDeploymentScale(t, namespace, appName, 0)
+
+	testlib.RequireEventually(t, func(requireEventually *require.Assertions) {
+		newPods := getRunningPodsByNamePrefix(t, namespace, appName+"-", ignorePodsWithNameSubstring)
+		requireEventually.Len(newPods, 0, "wanted zero pods")
+	}, 2*time.Minute, 200*time.Millisecond)
+
+	// Reset the application to its original scale.
+	updateDeploymentScale(t, namespace, appName, originalScale)
+
+	testlib.RequireEventually(t, func(requireEventually *require.Assertions) {
+		newPods := getRunningPodsByNamePrefix(t, namespace, appName+"-", ignorePodsWithNameSubstring)
+		requireEventually.Len(newPods, originalScale, "wanted %d pods", originalScale)
+	}, 2*time.Minute, 200*time.Millisecond)
+}
+
+// TestRemoveAllowedCiphersFromStaticConfig_Disruptive updates the Supervisor's static configuration to make sure that the allowed ciphers list is empty.
+// It will restart the Supervisor pods. Skipped because it's only here for local testing purposes.
+func TestRemoveAllowedCiphersFromStaticConfig_Disruptive(t *testing.T) {
+	t.Skip()
```

Comment:
Is this whole test basically dead code? Are we sure we want to commit dead code? I don't think I understood the use case where you might need this code.

Reply:
It's dead code that I used while making the actual integration test but it has no purpose now. Will be removed with the next push.